src/lib.rs

```diff
@@ -106,12 +108,70 @@ struct CorsHandlerAllowAny {
 
 impl CorsHandlerWhitelist {
     fn add_cors_header(&self, headers: &mut headers::Headers, origin: &headers::Origin) {
-        let header = match origin.host.port {
-            Some(port) => format!("{}://{}:{}", &origin.scheme, &origin.host.hostname, &port),
-            None => format!("{}://{}", &origin.scheme, &origin.host.hostname),
-        };
+        let header = format_cors_origin(origin);
         headers.set(headers::AccessControlAllowOrigin::Value(header));
     }
+
+    fn add_cors_preflight_headers(&self, 
+                                  headers: &mut headers::Headers, 
+                                  origin: &headers::Origin, 
+                                  acrm: &headers::AccessControlRequestMethod, 
+                                  acrh: Option<&headers::AccessControlRequestHeaders>) {
+        self.add_cors_header(headers, origin);
+
+        //Copy the method requested by the browser in the allowed methods header
+        headers.set(headers::AccessControlAllowMethods(vec!(acrm.0.clone())));
+        
+        //If we have special allowed headers, copy them in the allowed headers in the response
+        if let Some(acrh) = acrh {
+            headers.set(headers::AccessControlAllowHeaders(acrh.0.clone()));
+        }
+    }
+
+    fn process_possible_preflight(&self, req: &mut Request, origin: headers::Origin) -> IronResult<Response> {
+        // Verify origin header
+        let may_process = self.allowed_hosts.contains(&format_cors_origin(&origin));
+
+        if !may_process {
+            warn!("Got disallowed preflight CORS request from {}", &origin.host.hostname);
+            return Ok(Response::with((status::BadRequest, "Invalid CORS request: Origin not allowed")));
+        }
```

Comment:
@dbrgn Done! 
I updated the version in Cargo.toml to 0.5.2, because I guess that you would like to publish in cargo. 

The readme now says that preflight request are supported :)